tests/image/test_pixel_overlap.py

```diff
@@ -0,0 +1,287 @@
+import unittest
+
+import pytest
+from PIL import Image, ImageChops
+
+from generic_grader.image.pixel_overlap import build
+from generic_grader.utils.options import Options
+
+
+@pytest.fixture()
+def built_class():
+    """Provide the class built by the build function."""
+    return build(
+        Options(region_inner="path", region_outer="walls", mode="exactly", threshold=0)
+    )
+
+
+@pytest.fixture()
+def built_instance(built_class):
+    """Provide an instance of the built class."""
+    return built_class()
+
+
+def test_build_class_type(built_class):
+    """Test that the file_closed build function returns a class."""
+    assert issubclass(built_class, unittest.TestCase)
+
+
+def test_build_class_name(built_class):
+    """Test that the built_class has the correct name."""
+    assert built_class.__name__ == "TestPixelOverlap"
+
+
+def test_built_instance_type(built_instance):
+    """Test that the built_class returns instances of unittest.TestCase."""
+    assert isinstance(built_instance, unittest.TestCase)
+
+
+def test_instance_has_test_method(built_instance):
+    """Test that instances of the built_class have test method."""
+    assert hasattr(built_instance, "test_pixel_overlap_0")
+
+
+def test_doc_func(built_instance):
+    """Test that the doc_func function returns the correct docstring."""
+    docstring = built_instance.test_pixel_overlap_0.__doc__
+    assert (
+        "Check that the path from your `main` function when called as "
+        "`main()` has exactly 0 pixels in the walls." == docstring
+    )
+
```

Comment:
We should check docstrings that also have entries for the assignments. For e.g. - Star Pattern requires us to enter the number of points for the star and we want that to show up in the docstring.